Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -240,18 +240,19 @@ recalculateFareForDistance ServiceHandle {..} booking ride recalcDistance = do
   putDiffMetric merchantId fareDiff distanceDiff
   return (recalcDistance, finalFare, Just fareParams)
 
-isPickupDropOutsideOfThreshold :: (MonadThrow m, Log m, MonadTime m, MonadGuid m) => ServiceHandle m -> SRB.Booking -> DRide.Ride -> LatLong -> m Bool
+isPickupDropOutsideOfThreshold :: (MonadThrow m, Log m, MonadTime m, MonadGuid m) => ServiceHandle m -> SRB.Booking -> DRide.Ride -> LatLong -> m (Bool, Bool)
 isPickupDropOutsideOfThreshold ServiceHandle {..} booking ride tripEndPoint = do
   let mbTripStartLoc = ride.tripStartPos
   thresholdConfig <- findConfig >>= fromMaybeM (InternalError "TransportConfigNotFound")
   -- for old trips with mbTripStartLoc = Nothing we always recalculate fare
   case mbTripStartLoc of
-    Nothing -> pure True
+    Nothing -> pure (True, True)
     Just tripStartLoc -> do
       let pickupLocThreshold = metersToHighPrecMeters thresholdConfig.pickupLocThreshold
       let dropLocThreshold = metersToHighPrecMeters thresholdConfig.dropLocThreshold
       let pickupDifference = abs $ distanceBetweenInMeters (getCoordinates booking.fromLocation) tripStartLoc
       let dropDifference = abs $ distanceBetweenInMeters (getCoordinates booking.toLocation) tripEndPoint
+      let dropLocOutsideOfThreshold = dropDifference >= dropLocThreshold
```

Comment:
This is calculated twice, you can change the return type to `(pickupOutsideOfThreshold, dropOutsideThreshold)`. This shouldn't require any refactoring as this fn is used only once ig.